tests/drivers/watchdog/wdt_basic_api/src/test_wdt.c

```diff
@@ -62,14 +62,34 @@
 #include <ztest.h>
 #include "test_wdt.h"
 
-#ifdef DT_ALIAS_WATCHDOG0_LABEL
-#define WDT_DEV_NAME DT_ALIAS_WATCHDOG0_LABEL
-#else
-#ifdef CONFIG_WWDG_STM32
-#define WDT_DEV_NAME DT_LABEL(DT_INST(0, st_stm32_window_watchdog))
-#else
-#define WDT_DEV_NAME DT_WDT_0_NAME
+/*
+ * To use this test, either the devicetree's /aliases must have a
+ * 'watchdog0' property, or one of the following watchdog compatibles
+ * must have an enabled node.
+ */
+#if DT_HAS_NODE(DT_ALIAS(watchdog0))
+#define WDT_NODE DT_ALIAS(watchdog0)
+#elif DT_HAS_COMPAT(st_stm32_window_watchdog)
+#define WDT_NODE DT_INST(0, st_stm32_window_watchdog)
+#elif DT_HAS_COMPAT(st_stm32_watchdog)
+#define WDT_NODE DT_INST(0, st_stm32_watchdog)
+#elif DT_HAS_COMPAT(nordic_nrf_watchdog)
+#define WDT_NODE DT_INST(0, nordic_nrf_watchdog)
+#elif DT_HAS_COMPAT(espressif_esp32_watchdog)
+#define WDT_NODE DT_INST(0, espressif_esp32_watchdog)
+#elif DT_HAS_COMPAT(silabs_gecko_wdog)
+#define WDT_NODE DT_INST(0, silabs_gecko_wdog)
+#elif DT_HAS_COMPAT(nxp_kinetis_wdog32)
+#define WDT_NODE DT_INST(0, nxp_kinetis_wdog32)
```

Comment:
Same question here.

Reply:
This shouldn't be changing any functionality. If you look at the fixups I've removed, wdog32 is the only one currently supported. Therefore I would leave adding additional compatibles not currently supported out of scope of this PR, unless I am missing something.